testing/rabbitmq-c: update to version 0.8.0

nathanejohnson opened the pull request:
Also splitting package into -dev, -doc and -bin packages.

jirutka commented:
Please examine the [build log](https://travis-ci.org/alpinelinux/aports/builds/137696175) and read readme of the rabbitmq. They use cmake, but also maintain legacy make/autohells build system. You’re somehow trying to use both, that doesn’t make sense.

nathanejohnson commented:
I have cleaned up the messiness in the build logs, however I'm not sure what you mean by mixing / matching cmake and autoconf .  I also bumped the pkgrel but I'm not sure if that's required in the github workflow.

jirutka commented on testing/rabbitmq-c/APKBUILD:
```diff
@@ -1,27 +1,50 @@
 build() {
 	cd "$builddir"
-	./configure --prefix=/usr || return 1
+	rm -rf build && mkdir build && cd build
```
There’s no `build` directory in the tarball; why you’re removing directory that doesn’t exist?

nathanejohnson replied:
If you run the build successively, it needs to be removed.  The -f flag ensures it still returns exit status of 0 and goes to the next step.  This is more for me while I was getting the build right.

jirutka commented on testing/rabbitmq-c/APKBUILD:
```diff
@@ -1,27 +1,50 @@
-	./configure --prefix=/usr || return 1
+	rm -rf build && mkdir build && cd build
+	cmake -DCMAKE_INSTALL_PREFIX=/usr ..
 	make || return 1
```
You should execute `cmake`, or `make`, not both of them. rabbitmq-c prefers cmake.

nathanejohnson replied:
Again, see below.

jirutka commented on testing/rabbitmq-c/APKBUILD:
```diff
@@ -1,27 +1,50 @@
 	make DESTDIR="$pkgdir" install || return 1
+	cd "$pkgdir"/usr
+	if [ -d ./lib64 ]; then
+		mv lib64 lib
```
Why this?

nathanejohnson replied:
This installs to lib64 on x86_64 builds, which doesn't appear to be an appropriate location on alpine.

jirutka replied:
Yeah, that’s true, but this is very bad approach to solve it. cmake accepts variable `CMAKE_INSTALL_LIBDIR` that allows to define name or path of the lib directory (example: `cmake -DCMAKE_INSTALL_LIBDIR=lib`).

jirutka commented on testing/rabbitmq-c/APKBUILD:
```diff
@@ -1,27 +1,50 @@
 package() {
-	cd "$builddir"
+	cd "$builddir"/build
 	make DESTDIR="$pkgdir" install || return 1
```
Again, do you wanna use cmake build system, or legacy autotools/make build system? It doesn’t make sense to use both at once. Please read https://github.com/alanxz/rabbitmq-c#building-and-installing.

nathanejohnson replied:
See comments below in the thread

nathanejohnson commented:
The use of cmake and make are not mutually exclusive and, in fact, go hand in hand.  cmake generates Makefiles.  Automake also generates Makefiles, however I am not using automake with this APKBUILD.  I do, however, use make in conjunction with cmake.  As do most of the aports/main APKBUILD files that use cmake, for what it's worth.

If there is some rule about not using make with cmake, perhaps you could help me with a hint as to how to have the installation go into a fakeroot, as with make DESTDIR ?

jirutka commented:
I owe you an apology. You’re right about cmake vs. make, it’s used in many abuild in the same way. It seems that I kinda didn’t understand what exactly cmake is and rabbitmq-c’s readme has supported me in my wrong idea that these are mutually exclusive.

nathanejohnson commented:
No worries!  So at this point, I can ditch that rm -rf on the build dir, outside of that do you have any remaining concerns?

nathanejohnson commented:
One last thing before I rebase / push, do I need to bump pkgrel or keep it at 0?

nathanejohnson commented:
I added the -DCMAKE_INSTALL_LIBDIR , removed the rm -rf , and reset pkgrel to 0 (I assume that this only needs to be bumped after merge).  Hopefully this addresses your concerns.

jirutka commented:
Thanks. The last concern/question: I think that `rabbitmq-c-utils` would be better than `rabbitmq-c-bin`. What do you think? (I can change it before merging, not needed to update the PR.)

nathanejohnson commented:
That works for me.

jirutka commented:
Merged in 6af1dbb56d77f977fb0a3988940eaa628e068651, thanks!
